ckanext/datastore/backend/postgres.py

```diff
@@ -1529,7 +1536,12 @@ def search_sql(context, data_dict):
 
         results = context['connection'].execute(sql)
 
-        return format_results(context, results, data_dict)
+        if results.rowcount == rows_max + 1:
+            data_dict['records_truncated'] = True
+        else:
+            data_dict['records_truncated'] = False
```

Comment:
`data_dict['records_truncated'] = results.rowcount == rows_max + 1`

Reply:
or better yet, not add anything to the dict in the common case
```python
        if results.rowcount == rows_max + 1:
            data_dict['records_truncated'] = True
```

Reply:
why not